server/world/block_source.go

```diff
@@ -0,0 +1,12 @@
+package world
+
+import "github.com/df-mc/dragonfly/server/block/cube"
+
+// BlockSource represents a source for blocks. Blocks can be retrieved
+// and set in the block source.
+type BlockSource interface {
+	// Block returns the block at the given position in the block source.
+	Block(cube.Pos) Block
+	// SetBlock sets the block at the given position in the block source.
+	SetBlock(cube.Pos, Block, *SetOpts)
```

Comment:
I believe SetBlock isn't used anywhere in this package, so I think you could remove this method from the interface.